docs/guides/typescript.md

```diff
@@ -26,65 +26,54 @@ const useBearStore = create<BearState>()((set) => ({
 
   <br/>
 
-**TLDR**: Because state generic `T` is invariant.
+**TLDR**: Because we just provide the information for the returned type but no information for the parameter type.
 
 Consider this minimal version `create`:
 
 ```ts
 declare const create: <T>(f: (get: () => T) => T) => T
-
-const x = create((get) => ({
+const x = create(() => ({
   foo: 0,
-  bar: () => get(),
```

Comment:
I think this is required to reproduce the problem. Otherwise, it would have been easier.

Reply:
Bring back `bar: () => get()` and add some text about how the ts behavior related to real-world scenario